src/node/snapshotter.h

```diff
@@ -23,6 +23,13 @@ namespace ccf
   public:
     static constexpr auto max_tx_interval = std::numeric_limits<size_t>::max();
 
+    enum class Flag : uint8_t
+    {
+      FORCE_SNAPSHOT_AT_NEXT_SIGNATURE = 0x01
+    };
+
+    kv::Version forced_snapshot_after_idx = -1;
```

Comment:
I don't like this extra variable at all, but it's needed because `commit()` would force snapshots too early sometimes (at the signature just before the proposal is accepted, instead of at the next one). That index could probably be put into `next_snapshot_indices` somehow, with a flag to force it immediately instead of checking the snapshot interval.